source/agora/consensus/data/Block.d

```diff
@@ -824,3 +824,53 @@ version (unittest)
         return block;
     }
 }
+
+/// demonstrate signing two blocks at height 1 to reveal private node key
+unittest
+{
+    import agora.consensus.data.genesis.Test: GenesisBlock;
+    import agora.crypto.ECC: Scalar, Point;
+    import agora.crypto.Schnorr: multiSigSign, multiSigVerify, Pair, Sig;
+    import std.format;
+
+    Hash random_seed1 = "seed1".hashFull();
+    Hash random_seed2 = "seed2".hashFull();
+
+    // Generate two blocks at height 1
+    auto block1 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(2).map!(txb => txb.sign()).array(), 10, random_seed1);
+    auto block2 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(3).map!(txb => txb.sign()).array(), 10, random_seed2);
+
+    Scalar v = genesis_validator_keys[0].secret;
```

Comment:
This is the problematic part. This secret is not on the curve at all. Probably something to do with the recent crypto changes. If you change it to `Scalar v = Scalar.random();` it will work fine.

Reply:
Full example where it works:

```D
import std.stdio;

Scalar v = Scalar.random();
//Scalar v = genesis_validator_keys[0].secret;

Scalar rc = Scalar(hashMulti(v, "consensus.signature.noise", 0));
Scalar r = rc + Scalar(block1.header.height.hashFull());

Scalar c1 = Scalar(hashFull(block1));
Scalar c2 = Scalar(hashFull(block2));

// Sign with same r twice
Scalar s1 = r + (v * c1);
Scalar s2 = r + (v * c2);

Scalar sig = s1 - s2;

writefln("Is it: %s", v == (s1 - s2) * (c1 - c2).invert);
assert(v == (s1 - s2) * (c1 - c2).invert);
```

Reply:
The problem is that the `WellKnownKeys` have SecretKey values which are not valid Scalar values. I will create a PR to address that specifically first.